controllers/rhmi/rhmi_controller.go

```diff
@@ -1419,3 +1441,63 @@ func isInstallationOlderThan1Minute(installation *rhmiv1alpha1.RHMI) bool {
 	aMinuteAgo := time.Now().Add(-time.Minute)
 	return aMinuteAgo.After(installation.CreationTimestamp.Time)
 }
+
+func validateAddOnStsRoleArnParamaterPattern(client k8sclient.Client, namespace string) (bool, error) {
```

Comment:
And in this function there is significant overlap with `getStsRoleArn` from the pkg/products/cloudresources/reconciler.go
Here we could just call that `getStsRoleArn` function and check if it return error or empty value.

Reply:
Could we discuss please. We talked before that it will be useful for SRE to know exact issue with ARN parameter. So in ```preflightChecks(..)```  - it's carefully checked.  In  products/cloudresources/reconciler.go  -  this detailed check seems already  not needed (as it was already checked in preflightChecks, in rhmi_controller.go. This is the reason that I thought that in ```preflightChecks``` - it should be detailed check, and in ```reconciler.go``` in ```getStsRoleArn``` - just brief check (sorry if my explanation is a bit confusing)